Source/Core/VideoCommon/VertexLoaderX64.cpp

```diff
@@ -0,0 +1,453 @@
+	}
+
+	OpArg data = GetVertexAddr(ARRAY_POSITION, m_VtxDesc.Position);
+	ReadVertex(data, m_VtxDesc.Position, m_VtxAttr.PosFormat, m_VtxAttr.PosElements + 2, 3, m_VtxAttr.PosFrac, &m_native_vtx_decl.position);
+
+	if (m_VtxDesc.Normal)
+	{
+		static const u8 map[] = {7, 6, 15, 14, 30};
+		u8 scaling_exponent = map[m_VtxAttr.NormalFormat];
+		if (m_VtxDesc.Normal & MASK_INDEXED && m_VtxAttr.NormalElements && m_VtxAttr.NormalIndex3)
+		{
+			for (int i = 0; i < (m_VtxAttr.NormalIndex3 ? 3 : 1); i++)
+			{
+				data = GetVertexAddr(ARRAY_NORMAL, m_VtxDesc.Normal);
+				ReadVertex(data, m_VtxDesc.Normal, m_VtxAttr.NormalFormat, 3, 3, scaling_exponent, &m_native_vtx_decl.normals[i]);
+			}
+		}
+		else
+		{
+			data = GetVertexAddr(ARRAY_NORMAL, m_VtxDesc.Normal);
+			for (int i = 0; i < (m_VtxAttr.NormalElements ? 3 : 1); i++)
+				data.offset += ReadVertex(data, m_VtxDesc.Normal, m_VtxAttr.NormalFormat, 3, 3, scaling_exponent, &m_native_vtx_decl.normals[i]);
+		}
+
+		m_native_components |= VB_HAS_NRM0;
+		if (m_VtxAttr.NormalElements)
+			m_native_components |= VB_HAS_NRM1 | VB_HAS_NRM2;
+	}
+
+	const u64 col[2] = {m_VtxDesc.Color0, m_VtxDesc.Color1};
+	for (int i = 0; i < 2; i++)
+	{
+		if (col[i])
+		{
+			data = GetVertexAddr(ARRAY_COLOR + i, col[i]);
+			ReadColor(data, col[i], m_VtxAttr.color[i].Comp, m_VtxAttr.color[i].Elements);
+			m_native_components |= VB_HAS_COL0 << i;
+			m_native_vtx_decl.colors[i].components = 4;
+			m_native_vtx_decl.colors[i].enable = true;
+			m_native_vtx_decl.colors[i].offset = m_dst_ofs;
+			m_native_vtx_decl.colors[i].type = VAR_UNSIGNED_BYTE;
+			m_native_vtx_decl.colors[i].integer = false;
+			m_dst_ofs += 4;
+		}
+	}
+
+	const u64 tc[8] = {
+		m_VtxDesc.Tex0Coord, m_VtxDesc.Tex1Coord, m_VtxDesc.Tex2Coord, m_VtxDesc.Tex3Coord,
+		m_VtxDesc.Tex4Coord, m_VtxDesc.Tex5Coord, m_VtxDesc.Tex6Coord, m_VtxDesc.Tex7Coord,
+	};
+	for (int i = 0; i < 8; i++)
+	{
+		int elements = m_VtxAttr.texCoord[i].Elements + 1;
+		if (tc[i])
+		{
+			data = GetVertexAddr(ARRAY_TEXCOORD0 + i, tc[i]);
+			u8 scaling_exponent = m_VtxAttr.texCoord[i].Frac;
+			ReadVertex(data, tc[i], m_VtxAttr.texCoord[i].Format, elements, elements, scaling_exponent, &m_native_vtx_decl.texcoords[i]);
+			m_native_components |= VB_HAS_UV0 << i;
+		}
+		if (tm[i])
+		{
+			m_native_components |= (VB_HAS_TEXMTXIDX0 | VB_HAS_UV0) << i;
+			m_native_vtx_decl.texcoords[i].enable = true;
+			MOVZX(64, 8, scratch1, MDisp(src_reg, texmatidx_ofs[i]));
+			PXOR(XMM0, R(XMM0));
+			CVTSI2SS(XMM0, R(scratch1));
+			if (tc[i])
+			{
+				m_native_vtx_decl.texcoords[i].components = 3;
+				if (elements == 1)
+					m_dst_ofs += sizeof(float);
+				MOVSS(MDisp(dst_reg, m_dst_ofs), XMM0);
+				m_dst_ofs += sizeof(float);
+			}
+			else
+			{
+				m_native_vtx_decl.texcoords[i].components = 4;
+				m_native_vtx_decl.texcoords[i].offset = m_dst_ofs;
+				m_native_vtx_decl.texcoords[i].type = VAR_FLOAT;
+				m_native_vtx_decl.texcoords[i].integer = false;
```

Comment:
I'd move the m_native_vtx_decl.texcoords[i].enable also here. For tc[i] == true, it's alredy set.